### DISCUSSION TITLE
Pause pipeline option required in gocd.yaml file configuration

### DISCUSSION THREAD

**Discussion by MRajasekhar** [gocd/gocd#10398] [Q&A]
Does anyone know how to pause pipeline from gocd.yaml file configuration instead of API call ?

**Comment by chadlwilson** [comment]
I don't think that is possible, because the paused/unpaused state of a pipeline is not something declarative that can be locked down as part of the pipeline's configuration. It is part of the runtime state of the pipeline that is available for users to interact with. It would be come out-of-sync with the configuration and partially defeat the purpose of "pipelines-as-code".

I wonder what you are trying to achieve though?

**Comment by sriramnrn** [comment]
On Sat, May 7, 2022 at 6:01 PM Chad Wilson ***@***.***> wrote:

> I don't think that is possible, because the paused/unpaused state of a
> pipeline is not something declarative that can be locked down as part of
> the pipeline's configuration. It is part of the runtime state of the
> pipeline that is available for users to interact with. It would be come
> out-of-sync with the configuration and partially defeat the purpose of
> "pipelines-as-code".
>
> I wonder what you are trying to achieve though?
>

In case you want to pause a pipeline, there is an API for it:
https://api.gocd.org/current/#pause-a-pipeline

If you want to pause the pipeline when creating it via API (different from
pipelines-as-code which Chad has mentioned), there's a header per this doc:
https://api.gocd.org/current/#create-a-pipeline
